wild_lib/src/elf_writer.rs

```diff
@@ -282,9 +287,43 @@ impl SizedOutput {
             let mut section_buffers = split_output_into_sections(layout, &mut self.out);
             sort_eh_frame_hdr_entries(section_buffers.get_mut(output_section_id::EH_FRAME_HDR));
         }
+
+        if layout.args().build_id {
+            self.write_gnu_build_id_note(layout)?;
+        }
+        Ok(())
+    }
+
+    fn write_gnu_build_id_note<S: StorageModel>(&mut self, layout: &Layout<S>) -> Result {
+        let computed_build_id = self.compute_gnu_build_id_note();
+
+        let mut buffers = split_output_into_sections(layout, &mut self.out);
+        let e = LittleEndian;
+        let (note_header, mut rest) =
+            from_bytes_mut::<NoteHeader>(buffers.get_mut(output_section_id::NOTE_GNU_BUILD_ID))
+                .map_err(|_| anyhow!("Insufficient .note.gnu.build-id allocation"))?;
+        note_header.n_namesz.set(e, GNU_NOTE_NAME.len() as u32);
+        note_header.n_descsz.set(e, size_of::<GnuBuildId>() as u32);
+        note_header.n_type.set(e, NT_GNU_BUILD_ID);
+
+        let name_out = crate::slice::slice_take_prefix_mut(&mut rest, GNU_NOTE_NAME.len());
+        name_out.copy_from_slice(GNU_NOTE_NAME);
+
+        rest.copy_from_slice(&computed_build_id.as_slice());
+
         Ok(())
     }
 
+    #[instrument(skip_all, name = "Compute build ID")]
+    fn compute_gnu_build_id_note(&self) -> GnuBuildId {
+        let output_buffer = match &self.out {
```

Comment:
I think this can probably be simplified a bit here. `OutputBuffer` implements `Deref`. So you can probably just do something like `let output_buffer = &*self.out;` or, if you prefer, `self.out.deref()`.